### DISCUSSION TITLE
Improving technical documentation for integrations (Tip => use jQuery with Breeze)

### DISCUSSION THREAD

**Discussion by dellg5** [breezefront/community#223] [3. Integrations]
My team and I would like to thank you for the work you have done free of charge to provide a solution for Magento 2 performance without the headless implementation.
Currently developing under Magento 2.4.6 -p3 with Varnish, Cache warmer, ElasticSuite etc... it works with Breeze without any side effects.

With our hindsight regarding the implementation of Breeze and the integration of custom modules, we'd like to recommend that you flesh out your technical documentation, which only explains basic concepts without concrete examples.

For example, your documentation describes how to manage utilities. I follow the instructions, but it doesn't work... so I say to myself "well, I'll see how they did it" and then.... NOTHING
After a while, I managed to find my answer by searching through the Breeze vendors.
You don't say everything in your documentation, which is a pity.

If it can help people who want to use jQuery with Breeze libraries.
Add the Jquery file to the global import in the layout

```
<item name="jquery" xsi:type="array">
          <item name="path" xsi:type="string">My_Module/js/lib/jquery-3.7.1</item>
</item>
```

Change in file to use jQuery when needed and $ for cash
```
window.jQuery = window.$ = jQuery; ` with `window.jQuery = /*window.$ =*/ jQuery;

// Map over the $ in case of overwrite
	// _$ = window.$;

jQuery.noConflict = function( deep ) {
	// if ( window.$ === jQuery ) {
	// 	window.$ = _$;
	// }
```

**Comment by vovayatsyuk** [comment]
Hey!

Thanks for the feedback. What do you mean by "manage utilities"? Please point to a particular page you would like to improve.

As for jQuery, I'm not sure. Why do you need it? In my practice, most jQuery plugins are bloated and outdated. It's far better to use vanilla js libraries.

**Comment by arnoldmontiel01** [comment]
@dellg5

Thank you for the TIP, I'm trying to use an old library based on jquery 3.6.0 / jquery-ui, following in your steps I get a lot of errors in some breeze JS files, I don't understand exactly where I shoud do this:

window.jQuery = window.$ = jQuery; ` with `window.jQuery = /*window.$ =*/ jQuery;

// Map over the $ in case of overwrite
	// _$ = window.$;

jQuery.noConflict = function( deep ) {
	// if ( window.$ === jQuery ) {
	// 	window.$ = _$;
	// }

Could you give me a hand? thank you

**Reply by dellg5** [reply]
@arnoldmontiel01 

have you solved your problem with  jquery 3.6.0 / jquery-ui ? 

Create custom module for add all lib in module :   app/web/app/code/Module/Libraries/view/frontend/layout/breeze_default.xml 

```
<referenceBlock name="breeze.js">
            <arguments>
                <argument name="bundles" xsi:type="array">
                    <item name="default" xsi:type="array">
                        <item name="items" xsi:type="array">
                            <item name="jquery" xsi:type="array">
                                <item name="path" xsi:type="string">Custom_Lib/js/lib/jquery-3.7.1</item>
                            </item>
                            </item>
                            <item name="exemple" xsi:type="array">
                                <item name="path" xsi:type="string">Custom_Lib/js/lib/exemple</item>
                            </item>             
                            </item>
                        </item>
                    </item>
                </argument>
            </arguments>
```

at this end of file jquery3-7-1 search and copy this part

```
    // Map over jQuery in case of overwrite
     var _jQuery = window.jQuery; // ,

    // Map over the $ in case of overwrite
    // _$ = window.$;

    jQuery.noConflict = function (deep) {
        // if ( window.$ === jQuery ) {
        // 	window.$ = _$;
        // }

        if (deep && window.jQuery === jQuery) {
            window.jQuery = _jQuery;
        }

        return jQuery;
    };

   / Expose jQuery and $ identifiers, even in AMD
   // (trac-7102#comment:10, https://github.com/jquery/jquery/pull/557)
  // and CommonJS for browser emulators (trac-13566)

    if (typeof noGlobal === "undefined") {
        window.jQuery = jQuery;   // window.$ =*
    }

    return jQuery;
});
```
and you can use in phtml or js file in module (if u use require in module add deps:jquery)
```
<script type="text/javascript">
    require([
        'jquery'
    ], function ($) {

or 

define([
    'jquery',
    'jquery/ui',
    'slick'
], function ($, slick) {
    'use strict';
    $(document).ready(function() {
```
